### PR TITLE
Implement Samsung SDI ESS System (legacy)

### PR DESCRIPTION
**Pull request by Sn0w3y**
## Description

This PR introduces support for the Samsung ESS system, specifically targeting older models that lack remote UI capabilities. This update is crucial for enhancing the integration capabilities of our energy management solutions, allowing users with legacy systems to connect and manage their energy storage effectively.

### Key Features
- **Compatibility Enhancement:** Ensures our system is compatible with Samsung SDI Lithium-Ionen Speicher 3.6 kWh, a model which has discontinued support for remote user interfaces.
- **Data Integration:** Implements data fetching and handling tailored to the unique requirements of the Samsung ESS system, accommodating its existing data interfaces and operational parameters.
- **Error Handling:** Robust error handling to manage connectivity issues or data parsing errors, which are more prevalent in older systems.

### Motivation
This enhancement responds to requests from users who continue to utilize older ESS models and face challenges in integrating these systems into modern energy management setups. 

### Useful Links
- Product Overview: [Samsung SDI Lithium-Ionen Speicher 3.6 kWh](https://www.photovoltaik4all.de/speicher/samsung/samsung-sdi-lithium-ionen-speicher-3-6-kwh/SA23695)

### Testing
- Detailed testing has been conducted to ensure compatibility and robust functioning. The tests included scenarios to verify accurate data fetching and error handling.

### CODE REVIEW

File: io.openems.edge.ess.samsung/src/io/openems/edge/ess/samsung/charger/SamsungEssChargerImpl.java [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+@Designate(ocd = Config.class, factory = true)
+@Component(//
+		name = "PV-Inverter.Samsung", immediate = true, //
+		configurationPolicy = ConfigurationPolicy.REQUIRE)
+
+@EventTopics({ //
+		EdgeEventConstants.TOPIC_CYCLE_AFTER_PROCESS_IMAGE, //
+})
+public class SamsungEssChargerImpl extends AbstractOpenemsComponent implements SamsungEssCharger, ElectricityMeter,
+		OpenemsComponent, EventHandler, TimedataProvider, ManagedSymmetricPvInverter {
```
Comment (sfeilmeier) [inline io.openems.edge.ess.samsung/src/io/openems/edge/ess/samsung/charger/SamsungEssChargerImpl.java:48]: Why is this not an EssDcCharger?

**Review by sfeilmeier — APPROVED** [review]
It's not perfect, but good enought for a legacy implementation. Please see my comments for possible future improvements

**Comment by Sn0w3y** [comment]
Hi @sfeilmeier unfortunately now the ESS is not shown as an ESS anymore.

image

**Comment by sfeilmeier** [comment]
Are you sure the ESS-Component is working (e.g. in log). I changed the Factory-IDs, so it has to be configured again.

**Comment by Sn0w3y** [comment]
Yes, 100% sure :)

image

**Comment by Sn0w3y** [comment]
Simple Python Script for your to test:

import json
from http.server import BaseHTTPRequestHandler, HTTPServer
from urllib.parse import urlparse, parse_qs


class RequestHandler(BaseHTTPRequestHandler):
    def do_GET(self):
        # Parse the query parameters
        query_components = parse_qs(urlparse(self.path).query)

        # Check if the correct file parameter is present
        if self.path.startswith('/R3EMSAPP_REAL.ems') and query_components.get('file') == ['ESSRealtimeStatus.json']:
            response = {
                "ESSRealtimeStatus": {
                    "ColecTm": "20240616205306",
                    "PowerOutletPw": "0",
                    "GridPw": 0.04,
                    "UnitPrice": 0.00,
                    "ConsPw": 0.45,
                    "BtSoc": 83,
                    "PcsPw": 416.00,
                    "AbsPcsPw": 0.42,
                    "PvPw": 0.00,
                    "GridStusCd": "0",
                    "BtStusCd": "0",
                    "BtPw": 0.46,
                    "OperStusCd": "0",
                    "EmsOpMode": "0",
                    "RankPer": 0,
                    "ErrorCnt": 0
                }
            }

            # Send response status code
            self.send_response(200)
            # Send headers
            self.send_header('Content-type', 'application/json')
            self.end_headers()
            # Send the JSON response
            self.wfile.write(json.dumps(response).encode('utf-8'))
        else:
            # Send a 400 Bad Request response if the file parameter is missing or incorrect
            self.send_response(400)
            self.send_header('Content-type', 'application/json')
            self.end_headers()
            self.wfile.write(json.dumps({"error": "Invalid file parameter"}).encode('utf-8'))


def run(server_class=HTTPServer, handler_class=RequestHandler, port=80):
    server_address = ('', port)
    httpd = server_class(server_address, handler_class)
    print(f'Starting httpd server on port {port}')
    httpd.serve_forever()


if __name__ == "__main__":
    run()

**Pull request merged**